inc/site-health.php

```diff
@@ -150,14 +150,16 @@ function output_compatibility_status_table( $plugins, $output = true, $incompati
 		<tbody>
 		<?php
 
-		// Filter out incompatible plugins. This allows us to re-use the status table for different types of compatibility issues.
-		if ( $incompatible && ( ! isset( $plugins['plugin_compatibility'] ) || $plugins['plugin_compatibility'] !== 'incompatible' ) ) {
-			$plugins = array_filter( $plugins, function ( $plugins ) {
-				return $plugins['plugin_compatibility'] === 'incompatible';
-			} );
-		}
+		foreach ( $plugins as $plugin => $field ) {
+			// Filter out incompatible plugins. This allows us to re-use the status table for different types of compatibility issues.
+			$plugin_compatibility = isset( $plugin['plugin_compatibility'] ) ? $plugin['plugin_compatibility'] : false;
+			// var_dump( $plugin_compatibility, wp_list_pluck( $plugins, 'plugin_compatibility' ) );
+			if ( $incompatible && ( ! $plugin_compatibility || $plugin_compatibility !== 'incompatible' ) ) {
```

Comment:
```suggestion
			if ( $incompatible && ( ! $plugin_compatibility || $plugin_compatibility !== 'incompatible' ) ) {
```

change `$incompatible` to something clearer about what we're doing here.